packages/flutter_tools/test/general.shard/localizations/gen_l10n_test.dart

```diff
@@ -0,0 +1,47 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:file/file.dart';
+import 'package:file/memory.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/localizations/gen_l10n.dart';
+
+import '../../src/common.dart';
+
+void main() {
+  late FileSystem fs;
+  late BufferLogger logger;
+
+  setUp(() {
+    fs = MemoryFileSystem.test();
+    logger = BufferLogger.test();
+  });
+
+  testWithoutContext('AppResourceBundle throws if file contains non-string value', () {
```

Comment:
nit. could/should this be in https://github.com/flutter/flutter/blob/master/packages/flutter_tools/test/general.shard/generate_localizations_test.dart ?

Reply:
I was wondering where the non-integration tests for this stuff was. Fixed.

Reply:
your file name was better, FWIW